### PR TITLE
Add Rotation.to/from_matrix() methods to convert between quaternions and orientation matrices

### PR DESCRIPTION
**Pull request by hakonanes**
Signed-off-by: Håkon Wiik Ånes <hwaanes@gmail.com>

Release notes: Add Rotation.to/from_matrix() methods to convert between quaternions and orientation matrices.

To do:
- [x] More robust tests
- [x] Think of any edge cases or different input shapes (help wanted)

### CODE REVIEW

**Comment by pc494** [comment]
Using randomly generated quaternions and running them in a loop is sufficiently robust for me (in terms of testing) as long as we spot the edge cases.

**Comment by hakonanes** [comment]
Ready for review.

File: orix/tests/test_rotation.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# from diffpy.structure.spacegroups import
 from math import cos, sin, tan, pi
```
Comment (dnjohnstone) [inline orix/tests/test_rotation.py:2]: Don't think we need the commented out diffpy import?

Comment (hakonanes) [reply orix/tests/test_rotation.py]: Yup, thanks.
